fedimint-client/src/sm/executor.rs

```diff
@@ -390,181 +410,235 @@ impl ExecutorInner {
+                                                state.clone(),
+                                            ))
+                                            .await;
+                                            dbtx.insert_entry(
+                                                &InactiveStateKey::from_state(state.clone()),
+                                                &meta.into_inactive(),
+                                            )
+                                            .await;
+
+                                            let context = &
+                                                module_contexts
+                                                .get(&state.module_instance_id())
+                                                .expect("Unknown module");
+
+                                            let global_context = global_context_gen(
+                                                state.module_instance_id(),
+                                                state.operation_id(),
+                                            );
+                                            if new_state.is_terminal(context, &global_context) {
+                                                // TODO: log state machine id or something
+                                                debug!("State machine reached terminal state");
+                                                let k =
+                                                    InactiveStateKey::from_state(new_state.clone());
+                                                let v = ActiveState::new().into_inactive();
+                                                dbtx.insert_entry(&k, &v).await;
+                                                Ok(ActiveOrInactiveState::Inactive {
+                                                    dyn_state: new_state,
+                                                })
+                                            } else {
+                                                let k =
+                                                    ActiveStateKey::from_state(new_state.clone());
+                                                let v = ActiveState::new();
+                                                dbtx.insert_entry(&k, &v).await;
+                                                Ok(ActiveOrInactiveState::Active {
+                                                    dyn_state: new_state,
+                                                    meta: v,
+                                                })
+                                            }
+                                        })
+                                    },
+                                    None,
                                 )
-                                .await;
+                                .await
+                                .expect("autocommit should keep trying to commit (max_attempt: None) and body doesn't return errors");
```

Comment:
running this expect fixes rustfmt :shrug: 

I pushed to 6e7946ac4a, feel free to use that

Reply:
"running"? I don't understand. But thanks a lot!

Reply:
edited, I can't even blame auto correct on laptop :laughing: